master/src/main/scala/org/apache/celeborn/service/deploy/master/tags/TagsManager.scala

```diff
@@ -43,19 +43,27 @@ class TagsManager extends Logging {
 
     if (tags.isEmpty) {
       logWarning("No tags provided")
-      return new util.ArrayList[WorkerInfo]()
+      return Collections.emptyList()
     }
 
-    // TODO: Support multiple tags (CELEBORN-1642)
-    val tag = tags(0)
-    val workersForTag = tagStore.get(tag)
-    if (workersForTag == null) {
-      logWarning(s"Tag $tag not found in cluster")
-      return new util.ArrayList[WorkerInfo]()
+    var workersForTags: Option[JSet[String]] = None
+    tags.foreach { tag =>
+      val taggedWorkers = tagStore.getOrDefault(tag, Collections.emptySet())
+      workersForTags match {
+        case Some(w) =>
+          w.retainAll(taggedWorkers)
+        case _ =>
+          workersForTags = Some(taggedWorkers)
+      }
+    }
+
+    if (workersForTags.isEmpty) {
+      logWarning(s"No workers for tags: $tagExpr found in cluster")
+      return Collections.emptyList()
     }
 
     val workerTagsPredicate = new Predicate[WorkerInfo] {
```

Comment:
This predicate can be merged into the filter method like the following codes.

`workers.stream().filter(w=>workersForTags.get.contains(w.toUniqueId())).collect(Collectors.toList())`

Reply:
@FMX i tried this and couple of other things, but java8 was not happy. So had to go predicate route.

Reply:
Lost the old logs to trying it out once again here.

Reply:
This is the error –
```
[error] /home/runner/work/celeborn/celeborn/master/src/main/scala/org/apache/celeborn/service/deploy/master/tags/TagsManager.scala:65:29: missing parameter type
[error]     workers.stream().filter(w => workersForTags.get.contains(w.toUniqueId())).collect(
[error]                             ^
[error] one error found
```

Reply:
OK, I understand. Scala 2.11 doesn't support Java 8 lambda natively.